tests/core/test_core.py

```diff
@@ -89,22 +89,25 @@ def test_task_fail_duration(self, dag_maker):
             )
         dag_maker.create_dagrun()
         session = settings.Session()
-        try:
-            op1.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
-        except Exception:
-            pass
-        try:
+        op1.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
+        with suppress(AirflowTaskTimeout):
             op2.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
-        except Exception:
-            pass
         op1_fails = (
             session.query(TaskFail)
-            .filter_by(task_id='pass_sleepy', dag_id=dag.dag_id, execution_date=DEFAULT_DATE)
+            .filter(
+                TaskFail.task_id == 'pass_sleepy',
+                TaskFail.dag_id == dag.dag_id,
+                TaskInstance.execution_date == DEFAULT_DATE,
+            )
             .all()
         )
         op2_fails = (
             session.query(TaskFail)
-            .filter_by(task_id='fail_sleepy', dag_id=dag.dag_id, execution_date=DEFAULT_DATE)
+            .filter(
+                TaskFail.task_id == 'fail_sleepy',
+                TaskFail.dag_id == dag.dag_id,
+                TaskInstance.execution_date == DEFAULT_DATE,
+            )
```

Comment:
```suggestion
            .filter(
                TaskFail.task_id == 'fail_sleepy',
                TaskFail.dag_id == dag.dag_id,
                DagRun.execution_date == DEFAULT_DATE,
            )
```

Same for other calls above.